sdk/python/feast/infra/offline_stores/snowflake_source.py

```diff
@@ -263,6 +263,11 @@ def get_table_column_names_and_types(
                                 result.dtypes[column].name
                             ]
                         else:
+                            if len(result) > 0:
+                                max_value = result.iloc[0][0]
+                                if max_value is not None and len(str(max_value)) <= 18:
+                                    row["snowflake_type"] = "NUMBER"
+                                    continue
```

Comment:
It either has to be NUMBER32 or NUMBER64

Reply:
@adchia

Reply:
where does this limitation come from? Is it the snowflake online store that specifically runs into this issue? I didn't see any issues locally when materializing to Redis

Reply:
Are you sure you are using the inference capability? NUMBER does not exist in the mapping

Here is the mapping:
```def snowflake_type_to_feast_value_type(snowflake_type: str) -> ValueType:
    type_map = {
        "BINARY": ValueType.BYTES,
        "VARCHAR": ValueType.STRING,
        "NUMBER32": ValueType.INT32,
        "NUMBER64": ValueType.INT64,
        "NUMBERwSCALE": ValueType.DOUBLE,
        "DOUBLE": ValueType.DOUBLE,
        "BOOLEAN": ValueType.BOOL,
        "DATE": ValueType.UNIX_TIMESTAMP,
        "TIMESTAMP": ValueType.UNIX_TIMESTAMP,
        "TIMESTAMP_TZ": ValueType.UNIX_TIMESTAMP,
        "TIMESTAMP_LTZ": ValueType.UNIX_TIMESTAMP,
        "TIMESTAMP_NTZ": ValueType.UNIX_TIMESTAMP,
    }
    return type_map[snowflake_type]```

Reply:
The limitation comes from protobuf not being able to handle numbers larger than int64?